src/simulation.jl

```diff
@@ -108,15 +110,34 @@ Base.Expr(c::AllocVecCall) = begin
   hook_AVC_caching(c, resolved_form, c.code_target)
 end
 
-function hook_AVC_caching(c::AllocVecCall, resolved_form::Symbol, ::CPUTarget)
+# TODO: Should maybe have default CPU generation be Vector with PreallocTools being opt-in
+"""
```

Comment:
We should add a subsection to the docs at some point that collects docstrings similar to this and add a text section that explains how to add a new backend etc.

Reply:
We can do that once we've stabilized the hooking API and actually expose them to the user.